src/System.Windows.Forms/src/System/Windows/Forms/CheckBox.CheckBoxAccessibleObject.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using static Interop;
+
+namespace System.Windows.Forms
+{
+    public partial class CheckBox
+    {
+        public class CheckBoxAccessibleObject : ControlAccessibleObject
+        {
+            public CheckBoxAccessibleObject(CheckBox owner) : base(owner)
```

Comment:
```suggestion
            public CheckBoxAccessibleObject(CheckBox owner) : base((owner as CheckBox) ?? throw new BlaBlaException(nameof(Owner)))
```

Reply:
I've looked at the available exceptions, and I think it should be [`ArgumentException`](https://docs.microsoft.com/dotnet/api/system.argumentexception)

Reply:
I suggest to also add an error message, something like `Argument '{0}' cannot be converted to type '{1}'.`
Similar to https://github.com/dotnet/winforms/blob/7ccf509e64d7958c38dd577162fa9ee946132233/src/Microsoft.VisualBasic.Forms/src/Resources/SR.resx#L128-L130

```suggestion
            public CheckBoxAccessibleObject(CheckBox owner) 
                : base((owner as CheckBox) ?? throw new ArgumentException(string.Format(SR.Argument_InvalidValueType, nameof(Owner), typeof(CheckBox)), nameof(Owner)))
```

Reply:
Is there a reason why you don't want to let the compiler do the work of ensuring the type (by declaring the argument `CheckBox` and just doing a null check) ?

Reply:
Normally we would force `CheckBox owner`, however unlike many other accessibility API, this one is a [public API](https://docs.microsoft.com/dotnet/api/system.windows.forms.checkbox.checkboxaccessibleobject), and changing type would be a breaking change. Granted it won't properly work with any other type, but this seems to be a lesser breaking change.

Reply:
oh, didn't notice that, yes changing the signature of a public API will cause MethodNotFound when you run a 3.x program on the 5.0 runtime (a secenario that was declared as desirable in previous discussions)

Reply:
Haha, I made the same mistake 3 hours earlier, and then Mikhail alerted me to the same fact.

Reply:
There has to be a pretty important reason to break the API surface. I can't even come up with a hypothetical case where we would (security perhaps, but I can't imagine a way we couldn't address a security problem that requires us to change the surface area).